AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+            exit_code, output = waagent.RunGetOutput(cmd, chk_err, log_cmd)
+    else:
+        try:
+            output = subprocess.check_output(cmd, stderr = subprocess.STDOUT,
+                                             shell = True)
+            exit_code = 0
+        except subprocess.CalledProcessError as e:
+            exit_code = e.returncode
+            output = e.output
+
+    return exit_code, output.encode('utf-8').strip()
+
+
+def init_waagent_logger():
+    """
+    Initialize waagent logger
+    If waagent has not been imported, catch the exception
+    """
+    try:
+        waagent.LoggerInit('/var/log/waagent.log', '/dev/stdout', True)
+    except Exception as e:
+        print('Unable to initialize waagent log because of exception ' \
+              '{0}'.format(e))
+
+def waagent_log_info(message):
+    """
+    Log informational message, being cautious of possibility that waagent may
+    not be imported
+    """
+    if 'Utils.WAAgentUtil' in sys.modules:
+        waagent.Log(message)
+    else:
+        print('Info: {0}'.format(message))
+
+
+def waagent_log_error(message):
+    """
+    Log error message, being cautious of possibility that waagent may not be
+    imported
+    """
+    if 'Utils.WAAgentUtil' in sys.modules:
+        waagent.Error(message)
+    else:
+        print('Error: {0}'.format(message))
+
+
+def hutil_log_info(message):
+    """
+    Log informational message, being cautious of possibility that hutil may
+    not be imported and configured
+    """
+    if HUtilObject is not None:
+        HUtilObject.log(message)
+    else:
+        print('Info: {0}'.format(message))
+
+
+def hutil_log_error(message):
+    """
+    Log error message, being cautious of possibility that hutil may not be
+    imported and configured
+    """
+    if HUtilObject is not None:
+        HUtilObject.error(message)
+    else:
+        print('Error: {0}'.format(message))
+
+
+def log_and_exit(operation, exit_code = 1, message = ''):
+    """
+    Log the exit message and perform the exit
+    """
+    if exit_code is 0:
+        waagent_log_info(message)
+        hutil_log_info(message)
+        exit_status = 'success'
+    else:
+        waagent_log_error(message)
+        hutil_log_error(message)
+        exit_status = 'failed'
+
+    if HUtilObject is not None:
+        HUtilObject.do_exit(exit_code, operation, exit_status, str(exit_code),
+                            message)
+    else:
+        update_status_file(operation, str(exit_code), exit_status, message)
+        sys.exit(exit_code)
+
+def get_log_directory():
+    if HUtilObject is None:
+        handler_env = get_handler_env()
+        return str(handler_env['handlerEnvironment']['logFolder'])
+    else:
+        return HUtilObject._context._log_dir
+
+def start_attestation_client():
+    """
+    Start attestation process that performs periodic monitoring of attestion process.
+    :return: None
+
+    """
+    stop_process('attestation_client')    
```

Comment:
Why are we trying to stop the attestation_client? We could check if it is already running we can simply exit from there?